# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../xf86-input-mutouch/0001-Fix-calculation-of-coordinates-with-inverted-axes.patch
# Copyright (C) 2010 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

From 275bc1fc265604ba15264b7d915c1e5208842fa7 Mon Sep 17 00:00:00 2001
From: Fernando Vicente <fvicente@gmail.com>
Date: Fri, 16 Jan 2009 06:53:22 +0100
Subject: [PATCH 1/2] Fix calculation of coordinates with inverted axes

Signed-off-by: Julien Cristau <jcristau@debian.org>
Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 src/xf86MuTouch.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/xf86MuTouch.c b/src/xf86MuTouch.c
index 82c2a25..2d5cdb0 100644
--- a/src/xf86MuTouch.c
+++ b/src/xf86MuTouch.c
@@ -426,9 +426,9 @@ xf86MuTReadInput(LocalDevicePtr	local)
 	 * or we will feed X with quite bogus event positions.
 	 */
         if (priv->x_inverted)
-          cur_x = priv->max_x - cur_x;
+          cur_x = priv->max_x - cur_x + priv->min_x;
         if (priv->y_inverted)
-          cur_y = priv->max_y - cur_y;
+          cur_y = priv->max_y - cur_y + priv->min_y;
 	xf86PostMotionEvent(local_to_use->dev, TRUE, 0, 2, cur_x, cur_y);
 
 	/*
-- 
1.6.5.3

